test/src/gtest_mpihpx_main.cpp

```diff
@@ -74,15 +74,9 @@ GTEST_API_ int main(int argc, char** argv) {
   listeners.Append(new MPIListener(argc, argv, default_listener));
 
   // Initialize HPX
-  hpx::start(nullptr, argc, argv);
-  hpx::runtime* rt = hpx::get_runtime_ptr();
-  hpx::util::yield_while([rt]() { return rt->get_state() < hpx::state_running; });
-
-  // Run!
-  auto ret = hpx::threads::run_as_hpx_thread(test_main, argc, argv);
+  auto ret = hpx::init(test_main, argc, argv);
 
   // Tear-down
-  hpx::stop();
```

Comment:
@rasolca do you agree with this change?

I remember when I developed it that you preferred the previous one, but I cannot recall the reason. Something may have changed and it may be ok now, just for checking.

Reply:
Sorry, this will be handled in a separate PR, I just have a few branches and occasionally merge stuff into this one. This PR is dependent on a few others and will not be merged soon.

The change was suggested by @msimberg ,perhaps, he can comment?

Reply:
The only reason I suggest `hpx::init` in this case is that it's significantly simpler and does the same thing. The `yield_while` requirement was removed in 1.4.0 so that would already simplify things a bit more. You would want to use `start`, `run_as_hpx_thread`, `stop` if you need to spawn multiple functions and the start/stop and spawning of functions are separate.